src/constants.rs

```diff
@@ -21,6 +21,13 @@ pub const HISTORY_RECEIPTS_COUNT: usize = 20;
 // This constant defines how often we check, In seconds
 pub const LOBBY_FLUSH_INTERVAL: usize = 5;
 
-pub const OAUTH_REDIRECT_URL: &str = "http://127.0.0.1:3002/auth/authorized";
-pub const OAUTH_AUTH_URL: &str = "https://kev-kzg-ceremony.eu.auth0.com/authorize";
-pub const OAUTH_TOKEN_URL: &str = "https://kev-kzg-ceremony.eu.auth0.com/oauth/token";
+pub const SIWE_OAUTH_REDIRECT_URL: &str = "http://127.0.0.1:3000/auth/callback/siwe";
```

Comment:
For local testing, I found it easier to have the port be different to 3000, how were you testing it locally?

Reply:
I don't see how the different port changes anything? I hit the `login` link, I get redirected back to my app. Why would I want a different port here?

Reply:
By "hit" I mean I copy-paste it from the `auth_link` endpoint

Reply:
For the authorisation code flow, the app gives the client a link, the client logs into the ouath server and then the oauth server redirects the client to some URL with the auth code and csrf token as get parameters. The client then sends that auth code and csrf token back to the app, so that they can get an access token.

When developing locally, I (client) setup a server on port 3002 and then I (app) setup a server on port 3000. It was then clear that the oauth server in this code flow, gave the auth code to the client and the client gives it to the app.

Reply:
Hmmmm now that you mention it... isn't the whole architecture broken if we want to facilitate independent frontends? With the current solution, we'll have to whitelist each frontend for the purpose of redirect (because that's what the providers want). So we probably want the frontends to pass a redirect param in the `request_link` call and we encode that in the `state` field. So then the flow would become:
1. Frontend asks for a link
2. User clicks the link, goes to oauth provider
3. oauth provider redirects to the backend
4. the backend does the authentication and redirects back to the frontend

Otherwise we're not really able to facilitate alternative frontends?

Reply:
Yeah I thought we were just going to whitelist frontends.

For whitelisting frontends, I think it should be fine though:

- If you are running it on your local computer, we can whitelist 127.0.0.1:3002/auth for example

- If you are running it for a larger audience, then it makes sense that there is some sort of whitelist in place, so we atleast check that you are not doing something fishy before we give you the heads up to facilitate contributions

I don't imagine more than a handful of different frontends though

Reply:
For the redirect param, we can probably make the clients life easier by having them send a parameter which allows us to add the appropriate redirect URI